front/src/modules/timesStops/hooks/useTimesStopsTableData.ts

```diff
@@ -0,0 +1,248 @@
+import { useEffect, useState, useMemo } from 'react';
+
+import { keyBy } from 'lodash';
+import { useTranslation } from 'react-i18next';
+import { useSelector } from 'react-redux';
+
+import usePathOps from 'applications/operationalStudies/hooks/usePathOps';
+import { useScenarioContext } from 'applications/operationalStudies/hooks/useScenarioContext';
+import type { PathPropertiesFormatted } from 'applications/operationalStudies/types';
+import { matchOpRefAndOp } from 'applications/operationalStudies/utils';
+import type { SimulationResponseSuccess, TrackSection } from 'common/api/osrdEditoastApi';
+import { matchPathStepAndOp } from 'modules/pathfinding/utils';
+import { interpolateValue } from 'modules/simulationResult/helpers/utils';
+import type { SimulationSummary } from 'modules/timetableItem/types';
+import type { Train } from 'reducers/osrdconf/types';
+import { getDisplayOnlyPathSteps } from 'reducers/simulationResults/selectors';
+import { Duration } from 'utils/duration';
+
+import { getTrackReferenceLabel, getOperationalPointName } from '../helpers/utils';
+import type { ScheduleEntry, TimesStopsRowNew } from '../types';
+
+type BuildTableRowParams = {
+  id: string;
+  name?: string;
+  ch?: string;
```

Comment:
Can you use `secondaryCode` here instead of ch ? Thanks :pray: 
```suggestion
  secondaryCode?: string;
```